commatrix/generate.go

```diff
@@ -0,0 +1,236 @@
+package commatrix
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"sync"
+
+	"golang.org/x/sync/errgroup"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	clientutil "github.com/openshift-kni/commatrix/client"
+	"github.com/openshift-kni/commatrix/consts"
+	"github.com/openshift-kni/commatrix/debug"
+	"github.com/openshift-kni/commatrix/ss"
+	"github.com/openshift-kni/commatrix/types"
+)
+
+func GenerateSS(kubeconfig, customEntriesPath, customEntriesFormat, format string, env Env, deployment Deployment, destDir string) (*types.ComMatrix, []byte, []byte, error) {
+	cs, err := clientutil.New(kubeconfig)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	nodesList, err := cs.CoreV1Interface.Nodes().List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	ssComMat, ssOutTCP, ssOutUDP, err := generateSS(cs, nodesList)
```

Comment:
Why did you separate to generateSS func that  is used just once?

Reply:
yah but the function is so much long this is why i seperate

Reply:
I don't think 5 more lines are a reason to open another func with the same name, if you want to make the functions shorter you can split parts in it to functions for example the loop that's getting the ss from the node

Reply:
i did on function its better